Assertion failure in FeedbackQuestionSubmissionEditPageAction

null

Request Path /page 
Request Parameters {courseid::TCP-122_2015, fsname::Peer assessment } 
Stack Trace


java.lang.AssertionError: null
	at teammates.common.util.Assumption.fail(Assumption.java:58)
	at teammates.common.util.Assumption.assertTrue(Assumption.java:27)
	at teammates.common.util.Assumption.assertNotNull(Assumption.java:254)
	at teammates.common.util.Assumption.assertNotNull(Assumption.java:246)
	at teammates.ui.controller.FeedbackQuestionSubmissionEditPageAction.execute(FeedbackQuestionSubmissionEditPageAction.java:22)
	at teammates.ui.controller.Action.executeAndPostProcess(Action.java:332)
	at teammates.ui.controller.ControllerServlet.doPost(ControllerServlet.java:58)
	at teammates.ui.controller.ControllerServlet.doGet(ControllerServlet.java:37)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:617)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:717)
	at org.mortbay.jetty.servlet.ServletHolder.handle(ServletHolder.java:511)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1166)
	at teammates.storage.datastore.DatastoreFilter.doFilter(DatastoreFilter.java:28)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.appengine.tools.appstats.AppstatsFilter.doFilter(AppstatsFilter.java:142)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.utils.servlet.ParseBlobUploadFilter.doFilter(ParseBlobUploadFilter.java:125)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.runtime.jetty.SaveSessionFilter.doFilter(SaveSessionFilter.java:35)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.utils.servlet.JdbcMySqlConnectionCleanupFilter.doFilter(JdbcMySqlConnectionCleanupFilter.java:60)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.utils.servlet.TransactionCleanupFilter.doFilter(TransactionCleanupFilter.java:43)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at org.mortbay.jetty.servlet.ServletHandler.handle(ServletHandler.java:388)
	at org.mortbay.jetty.security.SecurityHandler.handle(SecurityHandler.java:216)
	at org.mortbay.jetty.servlet.SessionHandler.handle(SessionHandler.java:182)
	at org.mortbay.jetty.handler.ContextHandler.handle(ContextHandler.java:765)
	at org.mortbay.jetty.webapp.WebAppContext.handle(WebAppContext.java:418)
	at com.google.apphosting.runtime.jetty.AppVersionHandlerMap.handle(AppVersionHandlerMap.java:254)
	at org.mortbay.jetty.handler.HandlerWrapper.handle(HandlerWrapper.java:152)
	at org.mortbay.jetty.Server.handle(Server.java:326)
	at org.mortbay.jetty.HttpConnection.handleRequest(HttpConnection.java:542)
	at org.mortbay.jetty.HttpConnection$RequestHandler.headerComplete(HttpConnection.java:923)
	at com.google.apphosting.runtime.jetty.RpcRequestParser.parseAvailable(RpcRequestParser.java:76)
	at org.mortbay.jetty.HttpConnection.handle(HttpConnection.java:404)
	at com.google.apphosting.runtime.jetty.JettyServletEngineAdapter.serviceRequest(JettyServletEngineAdapter.java:146)
	at com.google.apphosting.runtime.JavaRuntime$RequestRunnable.run(JavaRuntime.java:527)
	at com.google.tracing.TraceContext$TraceContextRunnable.runInContext(TraceContext.java:437)
	at com.google.tracing.TraceContext$TraceContextRunnable$1.run(TraceContext.java:444)
	at com.google.tracing.CurrentContext.runInContext(CurrentContext.java:220)
	at com.google.tracing.TraceContext$AbstractTraceContextCallback.runInInheritedContextNoUnref(TraceContext.java:308)
	at com.google.tracing.TraceContext$AbstractTraceContextCallback.runInInheritedContext(TraceContext.java:300)
	at com.google.tracing.TraceContext$TraceContextRunnable.run(TraceContext.java:441)
	at com.google.apphosting.runtime.ThreadGroupPool$PoolEntry.run(ThreadGroupPool.java:251)
	at java.lang.Thread.run(Thread.java:745)

--- a/src/main/webapp/js/instructorFeedbackEdit.js
+++ b/src/main/webapp/js/instructorFeedbackEdit.js
@@ -607,7 +607,7 @@ function formatQuestionNumbers(){
 
 function getQuestionLink(qnNumber) {
     var courseid = $("input[name='courseid']").val();
-    var fsname = toParameterFormat($("input[name='fsname']").val());
+    var fsname = encodeURIComponent($("input[name='fsname']").val());
     
     var questionId = $("#form_editquestion-" + qnNumber)
                         .find("input[name='questionid']").val();
@@ -621,7 +621,7 @@ function getQuestionLink(qnNumber) {
     var questionLink =  window.location.protocol + "//" 
                         + window.location.host + actionUrl
                         + "?courseid=" + courseid 
-                        + "&fsname=" + replaceAll(fsname, "#", "%23")
+                        + "&fsname=" + fsname
                         + "&questionid=" + questionId;
     
     setStatusMessage("Link for question " + qnNumber + ": " + questionLink, false);

--- a/src/main/webapp/js/instructorFeedbackEdit.js
+++ b/src/main/webapp/js/instructorFeedbackEdit.js
@@ -621,7 +621,7 @@ function getQuestionLink(qnNumber) {
     var questionLink =  window.location.protocol + "//" 
                         + window.location.host + actionUrl
                         + "?courseid=" + courseid 
-                        + "&fsname=" + fsname
+                        + "&fsname=" + fsname 
                         + "&questionid=" + questionId;
     
     setStatusMessage("Link for question " + qnNumber + ": " + questionLink, false);

--- a/src/main/webapp/js/instructorFeedbackEdit.js
+++ b/src/main/webapp/js/instructorFeedbackEdit.js
@@ -621,7 +621,7 @@ function getQuestionLink(qnNumber) {
     var questionLink =  window.location.protocol + "//" 
                         + window.location.host + actionUrl
                         + "?courseid=" + courseid 
-                        + "&fsname=" + fsname 
+                        + "&fsname=" + replaceAll(fsname, "#", "%23")
                         + "&questionid=" + questionId;
     
     setStatusMessage("Link for question " + qnNumber + ": " + questionLink, false);

--- a/src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackEditPageUiTest.java
+++ b/src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackEditPageUiTest.java
@@ -245,7 +245,7 @@ public class InstructorFeedbackEditPageUiTest extends BaseUiTestCase {
         String expectedUrl = TestProperties.inst().TEAMMATES_URL
                 + Const.ActionURIs.INSTRUCTOR_FEEDBACK_QUESTION_SUBMISSION_EDIT_PAGE
                 + "?courseid=" + courseId
-                + "&fsname=First+Session"
+                + "&fsname=First%20Session"
                 + "&questionid=" + questionId;
 
         assertTrue(feedbackEditPage.isElementVisible("statusMessage"));

